yt/frontends/enzo/data_structures.py

```diff
@@ -370,20 +370,21 @@ def _detect_active_particle_fields(self):
                 self.dataset.particle_types = new_ptypes
                 self.dataset.particle_types_raw = new_ptypes
                 continue
-            gs = self.grids[select_grids > 0]
-            g = gs[0]
-            handle = h5py.File(g.filename, mode="r")
-            node = handle["/Grid%08i/Particles/" % g.id]
-            for ptype in (str(p) for p in node):
-                if ptype not in _fields:
-                    continue
-                for field in (str(f) for f in node[ptype]):
-                    _fields[ptype].append(field)
-                    sh = node[ptype][field].shape
-                    if len(sh) > 1:
-                        self.io._array_fields[field] = (sh[1],)
-                fields += [(ptype, field) for field in _fields.pop(ptype)]
-            handle.close()
+            if ptype != "DarkMatter":
+                gs = self.grids[select_grids > 0]
+                g = gs[0]
+                handle = h5py.File(g.filename, "r")
+                node = handle["/Grid%08i/Active Particles/" % g.id]
```

Comment:
The failing test is because our sample data has a different name for the active particles in the file. The code below made it work for me.
```suggestion
                grid_group = handle[f"/Grid{g.id:08d}"]
                found = False
                for pname in ["Active Particles", "Particles"]:
                    if pname in grid_group:
                        found = True
                        break
                if not found:
                    raise RuntimeError(
                        "Could not find active particle group in data.")
                node = grid_group[pname]
```